compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/DecomposeLinalgExtPackUnPackToAIR.cpp

```diff
@@ -0,0 +1,864 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "air/Dialect/AIR/AIRDialect.h"
+#include "iree-amd-aie/Transforms/PassDetail.h"
+#include "iree-amd-aie/Transforms/Passes.h"
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtDialect.h"
+#include "iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.h"
+#include "mlir/Dialect/Arith/Utils/Utils.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/Utils/IndexingUtils.h"
+#include "mlir/Dialect/Utils/StaticValueUtils.h"
+#include "mlir/IR/AffineMap.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+//===----------------------------------------------------------------------===//
+// Pass
+//===----------------------------------------------------------------------===//
+
+namespace mlir::iree_compiler::AMDAIE {
+
+namespace {
+
+/// Returns the padding value if set. Otherwise, returns the source directly.
+/// The method assumes that the `packOp` has static shapes.
+static Value getPackOpInputOrPaddedSource(OpBuilder &builder,
+                                          IREE::LinalgExt::PackOp packOp) {
+  return packOp.getInput();
+}
+
+OpFoldResult getMixedSize(OpBuilder &builder, Location loc, Value value,
+                          int64_t dim) {
+  auto memrefType = llvm::cast<MemRefType>(value.getType());
+  SmallVector<OpFoldResult> result;
+  // TODO: dynamic shape
+  // if (memrefType.isDynamicDim(dim))
+
+  return builder.getIndexAttr(memrefType.getDimSize(dim));
+}
+
+SmallVector<OpFoldResult> getMixedSizes(OpBuilder &builder, Location loc,
+                                        Value value) {
+  auto memrefType = llvm::cast<MemRefType>(value.getType());
+  SmallVector<OpFoldResult> result;
+  for (int64_t i = 0; i < memrefType.getRank(); ++i)
+    result.push_back(getMixedSize(builder, loc, value, i));
+  return result;
+}
```

Comment:
Does these upstream methods work?

https://github.com/llvm/llvm-project/blob/02f95b77515fe18ed1076b94cbb850ea0cf3c77e/mlir/include/mlir/Dialect/MemRef/IR/MemRef.h#L66-L68

Reply:
Thanks for the pointer. Yes the upstream method works. PR updated.